recipe/2025-09-container.md

```diff
@@ -0,0 +1,836 @@
+# Container Resource Recipe Migration
+
+* **Author**: Brooke Hamilton (@Brooke-Hamilton)
+
+## Overview
+
+This design describes replacing the imperative Go renderer chain for Applications.Core/containers with a Bicep recipe for the new Radius.Compute/containers resource type. The current implementation uses a chain of wrapper renderers (`kubernetesmetadata.Renderer` → `manualscale.Renderer` → `daprextension.Renderer` → `container.Renderer`) that generates Kubernetes Deployments, Services, Secrets, RBAC resources, and Azure identity resources. This will be replaced with a single Bicep recipe that works with the new resource schema.
+
+## Terms and definitions
+
+- **Container Renderer Chain**: The composed renderers in `pkg/corerp/model/application_model.go` that process containers
+- **Bicep Recipe**: Declarative template that provisions infrastructure based on input parameters
+- **Extension Renderers**: Wrapper renderers that add functionality (Dapr sidecars, manual scaling, metadata)
+- **Base Manifest**: User-provided Kubernetes YAML merged with generated resources via `kubeutil.ParseManifest`
+- **LocalID**: Internal identifier for tracking output resource relationships (e.g., `rpv1.LocalIDDeployment`)
+
+## Objectives
+
+### Goals
+
+- Replace the container renderer chain with a Bicep recipe for the new Radius.Compute/containers resource type
+- Implement all current functionality for the new resource schema: multi-container deployments, volumes, identity, RBAC, connections
+- Enable platform engineers to customize container deployment through recipe modifications
+- Align with the new resource type schema defined in resource-types-contrib
+
+### Non-goals
+
+- Supporting the old Applications.Core/containers resource type
+- Modifying the recipe engine or deployment processor  
+- Supporting non-Kubernetes platforms in this phase
+- Maintaining compatibility with existing deployed Applications.Core/containers resources
+
+## User Experience
+
+Users will define Radius.Compute/containers resources using the new schema:
+
+```bicep
+resource myContainer 'Radius.Compute/containers@2025-08-01-preview' = {
+  name: 'myContainer'
+  properties: {
+    environment: environment
+    application: application.id
+    containers: {
+      web: {
+        image: 'nginx:latest'
+        ports: {
+          http: {
+            containerPort: 80
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The recipe produces the same Kubernetes resources as the current renderer chain but works with the new resource type schema.
+
+## Bicep Kubernetes Extension Configuration
```

Comment:
maybe I'm missing something but this feels out of place. Is this implementation detail or user experience?